robot/Cumulus/perftests/BDI_Tests.robot

```diff
@@ -1,22 +1,138 @@
-                # half (num_records) will be "not matching"
-                #
-                # learn more here:
-                # https://salesforce.quip.com/gLfGAPtqVzUS
+${data_generation_task} =       tasks.generate_bdi_data.GenerateBDIData
 
 *** Settings ***
 
 Resource  cumulusci/robotframework/CumulusCI.robot
 Resource        robot/Cumulus/resources/NPSP.robot
-Suite Setup       Run Task Class   tasks.generate_bdi_data.GenerateBDIData
-...            num_records=${count/2}  mapping_yaml=datasets/bdi_benchmark/mapping.yml
+Resource        robot/Cumulus/resources/Data.robot
+Resource        robot/Cumulus/resources/BDI_API.robot
+Suite Setup       Workaround Bug
 
-*** Test Cases ***
+*** Keywords ***
+Clear Generated Records
+    Output  Clearing Generated Records
+    # Organized in dependency order
+    Delete     DataImport__c, CustomObject1__c, CustomObject2__c, CustomObject3__c      
+    Delete     Account_Soft_Credit__c     where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Account__r.Name Like '%BDITEST%'
+    Delete     Allocation__c        where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Opportunity__r.Account.Name Like '%BDITEST%'
+    Delete     npe01__OppPayment__c     where=npe01__Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR npe01__Opportunity__r.Account.Name Like '%BDITEST%'
+    Delete     Opportunity     where=Primary_Contact__r.LastName Like '%BDITEST%' OR Account.Name Like '%BDITEST%'
+    Delete     Account     where=Name Like '%BDITEST%'
+    Delete     Contact     where=LastName Like '%BDITEST%'
+
+Generate Data
+    [Arguments]    ${count}
+    ${count} =  Convert To Integer	${count}
+
+    Run Task Class   tasks.generate_and_load_data.GenerateAndLoadData
+    ...                 num_records=${count}
+#    ...                 database_url=sqlite:////tmp/temp_db.db  # turn this on to look at the DB for debugging
+    ...                 mapping=datasets/bdi_benchmark/mapping.yml
+    ...                 data_generation_task=${data_generation_task}
+
+Setup BDI
+    [Arguments]     ${field_mapping_method}
+    Configure BDI     ${field_mapping_method}
+
+    Run Keyword If    '${field_mapping_method}'=='Data Import Field Mapping'
+    ...               Ensure Custom Metadata Was Deployed
+
+Report BDI
+    ${result} =   Row Count      DataImport__c  
+    ...           Status__c=Imported
+
+    Output  DataImport__c imported    ${result}
+
+    ${result} =   Row Count     CustomObject3__c  
+
+    Output  CustomObject3__c imported    ${result}
+
+    ${result} =   Row Count  Account  
+    ...           BillingCountry=Tuvalu
+
+    Output  Accounts imported    ${result}
+
+    ${result} =   Row Count     Contact  
+    ...           Name Like '%BDITEST%'
+
+    Output  Contacts imported    ${result}
 
-Import a data batch via the API
+Workaround Bug
+    Return From Keyword If      ${persistent_org}   # persistent orgs don't have this bug
```

Comment:
Can you add a bit more of a comment about what bug is being worked around?

Reply:
https://github.com/SalesforceFoundation/Cumulus/pull/4592/files#diff-c514c9a06a353c336a1da458456bac85R64